File: c/2.c
#include <stdio.h>

int main()
{
    int a, b, c, total;
    a = 1;
    b = 1;
    total = 0;

    while (b < 4000000) {
        c = b;
        b = a + b;
        a = c;
        if (b % 2 == 0) {
            total += b;
        }
    }
    printf("%d\n", total);
}

File: rust/src/bin/2.rs
fn main() {
    let mut a = 1;
    let mut b = 1;
    let mut total = 0;

    while b < 4000000 {
        let c = b;
        b += a;
        a = c;
        if b % 2 == 0 {
            total += b;
        }
    }
    println!("{}", total)
}
